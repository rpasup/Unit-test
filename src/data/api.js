import data from './productdata.json';

export const getAllProducts = () => {
    return data && data.products;
}