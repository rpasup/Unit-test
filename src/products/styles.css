table {
  margin-left: auto;
  margin-right: auto;
}
table,
th,
td {
  border: 1px solid black;
  border-spacing: 5px;
}
th,
td {
  padding: 15px;
}

h3 {
  text-align: left;
}
