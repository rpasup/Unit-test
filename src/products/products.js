import React from "react";
import Product from "./product/product";
import "./styles.css";

export const Products = ({ products }) => {
  return (
    <div className="products">
      <h1>Product List</h1>
      <table>
        <thead>
          <tr>
            <th>{"ID"}</th>
            <th>{"Product Name"}</th>
            <th>{"Quantity"}</th>
            <th>{"Price"}</th>
          </tr>
        </thead>
        <tbody>
          {products &&
            products.map((prod) => <Product key={prod.id} product={prod} />)}
        </tbody>
      </table>
    </div>
  );
};
