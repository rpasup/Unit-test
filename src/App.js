import React from "react";
import "./App.css";
import { getAllProducts } from "./data/api";
import { Products } from "./products/products";

function App() {
  return (
    <div className="App">
      <Products products={getAllProducts()} />
    </div>
  );
}

export default App;
