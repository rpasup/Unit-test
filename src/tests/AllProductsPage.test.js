import React from "react";
import { Products } from "../products/products";
import { getAllProducts } from "../data/api";
import App from "../App";
import { shallow } from "enzyme";
import Product from "../products/product/product";

describe("All Products", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  test("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe("Test case to check if correct heading name is rendered", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Products products={getAllProducts()} />);
  });

  test("and it renders correctly", () => {
    expect(wrapper.find("h1").render().text()).toEqual("Product List");
  });
});

describe("Test case to check if one ProductList React component is rendered", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Products products={getAllProducts()} />);
  });

  test("and it renders correctly", () => {
    expect(wrapper.find(Product).length).toBeGreaterThan(1);
  });
});
