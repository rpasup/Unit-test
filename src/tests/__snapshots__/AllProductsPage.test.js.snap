// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All Products renders correctly 1`] = `
<div
  className="App"
>
  <Products
    products={
      Array [
        Object {
          "id": 1,
          "price": 13000,
          "productName": "Moto G5",
          "quantity": 2,
        },
        Object {
          "id": 2,
          "price": 6000,
          "productName": "Racold Geyser",
          "quantity": 3,
        },
        Object {
          "id": 3,
          "price": 50000,
          "productName": "Dell Inspiron",
          "quantity": 4,
        },
      ]
    }
  />
</div>
`;
